static/app/views/projectsDashboard/index.tsx

```diff
@@ -81,6 +81,53 @@ function addProjectsToTeams(teams: Team[], projects: Project[]): TeamWithProject
   }));
 }
 
+function getFilteredProjectsBasedOnTeams({
+  allTeams,
+  userTeams,
+  selectedTeams,
+  isAllTeams,
+  showNonMemberProjects,
+  projects,
+  projectQuery,
+}: {
+  allTeams: Team[];
+  isAllTeams: boolean;
+  projectQuery: string;
+  projects: Project[];
+  selectedTeams: string[];
+  showNonMemberProjects: boolean;
+  userTeams: Team[];
+}): Project[] {
+  const myTeamIds = new Set(userTeams.map(team => team.id));
+  const includeMyTeams = isAllTeams || selectedTeams.includes('myteams');
+  const selectedOtherTeamIds = new Set(
+    selectedTeams.filter(teamId => teamId !== 'myteams')
+  );
+  const myTeams = includeMyTeams ? allTeams.filter(team => myTeamIds.has(team.id)) : [];
+  const otherTeams = isAllTeams
+    ? allTeams
+    : [...allTeams].filter(team => selectedOtherTeamIds.has(String(team.id)));
+
+  const visibleTeams = [...myTeams, ...otherTeams].filter(team => {
+    if (showNonMemberProjects) {
+      return true;
+    }
+    return team.isMember;
+  });
+  const teamsWithProjects = addProjectsToTeams(visibleTeams, projects);
+  const currentProjects = uniqBy(
+    teamsWithProjects.flatMap(team => team.projects),
+    'id'
+  );
+  const currentProjectIds = new Set(currentProjects.map(p => p.id));
+  const unassignedProjects = isAllTeams
+    ? projects.filter(project => !currentProjectIds.has(project.id))
```

Comment:
does this need to check showNonMemberProjects? Orgs that have open team membership disabled may not want to show them

Reply:
But in this case, shouldn't other members be aware of these teams, so they can decide whether or not to request to join? or how they would ever know about these teams?

Reply:
in the settings, they can see all teams and request to join them:

![image](https://github.com/user-attachments/assets/3a943eca-8c4c-49b0-89e7-2b682b48c759)

Reply:
i think settings is the only place they should see them with open membership closed. we don't want them to see stats etc